core/src/main/java/org/apache/calcite/runtime/ResultSetEnumerable.java

```diff
@@ -47,6 +49,8 @@
   private final DataSource dataSource;
   private final String sql;
   private final Function1<ResultSet, Function0<T>> rowBuilderFactory;
+  private final Consumer<PreparedStatement> consumer;
```

Comment:
It should better be called `dynamicParamBinder` (or with another name).
`consumer` obscures the intention.

Reply:
The type Consumer<PreparedStatement> should also be changed to specific. Consumer urges to eat exceptions in accet(PreparedStatement).

Reply:
A full blown class is good as well. It could be nested to shave some code lines on license/end headers.